django-rgd-imagery/rgd_imagery/serializers/processed.py

```diff
@@ -1,59 +1,61 @@
+import logging
+
 from rest_framework import serializers
-from rest_framework.reverse import reverse
 from rgd.permissions import check_write_perm
+from rgd.serializers import ChecksumFileSerializer
 
 from .. import models
 from .base import ImageSerializer
 
+logger = logging.getLogger(__name__)
+
 
-class ConvertedImageSerializer(serializers.ModelSerializer):
+class ProcessedImageSerializer(serializers.ModelSerializer):
 
-    processed_image = ImageSerializer(read_only=True)
+    processed_image = ImageSerializer(required=False)
+    ancillary_files = ChecksumFileSerializer(many=True, required=False)
 
     def validate_source_image(self, value):
         if 'request' in self.context:
             check_write_perm(self.context['request'].user, value)
         return value
 
     class Meta:
-        model = models.ConvertedImage
+        model = models.ProcessedImage
         fields = '__all__'
-        read_only_fields = ['id', 'status', 'failure_reason', 'processed_image']
+        read_only_fields = [
+            'id',
+            'status',
+            'failure_reason',
+        ]
 
+    def create(self, validated_data):
+        """Prevent duplicated subsamples from being created."""
+        obj, created = models.ProcessedImage.objects.get_or_create(**validated_data)
+        if not created:
+            # Trigger save event to reprocess the subsampling
+            obj.save()
+        return obj
 
-class RegionImageSerializer(serializers.ModelSerializer):
 
-    processed_image = ImageSerializer(read_only=True)
+class ProcessedImageGroupSerializer(serializers.ModelSerializer):
 
     def validate_source_image(self, value):
         if 'request' in self.context:
             check_write_perm(self.context['request'].user, value)
         return value
 
-    def to_representation(self, value):
-        ret = super().to_representation(value)
-        realtive_status_uri = reverse('subsampled-status', args=[value.id])
-        if 'request' in self.context:
-            request = self.context['request']
-            ret['status'] = request.build_absolute_uri(realtive_status_uri)
-        else:
-            ret['status'] = realtive_status_uri
-        return ret
-
     class Meta:
-        model = models.RegionImage
-        fields = '__all__'
+        model = models.ProcessedImageGroup
+        fields = [
+            'parameters',
+        ]
         read_only_fields = [
-            'id',
-            'status',
-            'failure_reason',
-            'processed_image',
+            'source_images',
         ]
 
     def create(self, validated_data):
-        """Prevent duplicated subsamples from being created."""
-        obj, created = models.RegionImage.objects.get_or_create(**validated_data)
-        if not created:
-            # Trigger save event to reprocess the subsampling
-            obj.save()
+        # TODO: why isn't this used by the creat view?
```

Comment:
@mcovalt, I can't track down why the create view endpoint isn't using this serializer. Any help would be appreciated.

See `CreateProcessedImageGroup` in `django-rgd-imagery/rgd_imagery/rest/post.py`

Reply:
Looks like the imports were missing. Would the changes I suggested resolve it?

Reply:
That did it! Thanks! I would have thought that setting `serializer_class` in the view would be enough